### ISSUE TITLE
Name change prompt

### ISSUE THREAD

**Issue by Allerek** [multitheftauto/mtasa-blue#2103]
**Is your feature request related to a problem? Please describe.**
Yeah, a lot of players dont change their default names.

**Describe the solution you'd like**
Add a prompt at a first launch saying that its highly appreciated to change your name in settings.

**Describe alternatives you've considered**
Maybe 'isPlayerNameRandomized' function?

**Comment by patrikjuvonen** [comment]
I somehow remember we had this in some ancient MTA version.

**Comment by Allerek** [comment]
> I somehow remember we had this in some ancient MTA version.
Afaik 0.5 had this

**Comment by Pirulax** [comment]
Why is it a problem if they dont change their name?

**Comment by AlexTMjugador** [comment]
> Why is it a problem if they dont change their name?

I guess it makes player identification more difficult for server admins, and people usually expect that nicknames are somewhat personal and unique. Auto-generated nicknames like "FlyingLobster87" do not fulfill those expectations, although I find them somewhat funny :)

I support showing a prompt the first time a user opens MTA: SA that remembers them to set their nickname to something more appropriate, although I'd still keep the nickname generation if they choose not to change it. I'd also get rid of wording like "highly appreciated", as the expectations I've talked about may matter more or less depending on the servers you play; for instance, I doubt that someone in a freeroam server with no rules really cares about generated nicknames, but in a RP server a proper nick is more important. I'd use words like "personalization".

**Comment by Dutchman101** [comment]
There's not much substance to the entire subject, but what there is:

- Random nicks are funny, as already mentioned
- Random nicks help players quickly identify newbies and players that have probably just installed MTA and started playing (as players play for longer, they naturally tend to change their nick at some point). This can serve purposes ranging from fun and trolling, to reaching out and helping them
- Players that are identified as newbie are often funny by themselves (not limited to the randomly generated name being funny) and identification as such increases the perceived fun by other players

None of these points are significant, they are just small aspects experienced by some players. But in my opinion this suggestion neither serves a significant or important purpose, so then it's not outweighed and probably not worth implementing.

@AlexTMjugador RPG and more serious servers that experience the downsides of randomly generated nicknames, can just add a welcome screen or prompt, where the player is asked to pick a specific nickname.

**Comment by AlexTMjugador** [comment]
> @AlexTMjugador RPG and more serious servers that experience the downsides of randomly generated nicknames, can just add a welcome screen or prompt, where the player is asked to pick a specific nickname.

Yeah, and AFAIK nowadays most servers desire to manage their own player accounts, adding to your point that this is an extremely minor suggestion.

Anyway, just for the record, if someone wants so badly to detect auto-generated nicknames on their server, [the code that generates them](https://github.com/multitheftauto/mtasa-blue/blob/master/Client/core/CNickGen.cpp) is pretty straightforward to understand even if you don't write C++, and a script to check whether the nickname is one of those is easy to do. That code hasn't changed in a meaningful way in >5 years, so I'd say it's likely so stay like that for the foreseeable future.

**Comment by PlatinMTA** [comment]
I don't think this is a problem but it could be a nice addition. First time opening MTA, coming from RAGE, SAMP or w/e and you get a pop out to change your name or keep the randomly generated one, or better yet, randomly generate another. That would be cool for fresh instalations. It would also let you know that you can probably customize a lot of your MTA, which you can.

Having a function to let you know if the name was randomly generated however is kinda useless imho.

**Comment by Allerek** [comment]
In general, short(2-3 steps) tutorial about MTA could be cool. A lot of people doesnt even know we have GUI skins in MTA, or that you can adjust your chat.
(Also, we could create small repository under skins.mtasa.com for people to publish their skins for MTA GUI)

**Comment by PlatinMTA** [comment]
> (Also, we could create small repository under skins.mtasa.com for people to publish their skins for MTA GUI)

I think this will be added on the new community site (just speculation, i don't know)
